bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/model/jsonadapter/EvaluationLatencyMetricsEventAdapterFactory.kt

```diff
@@ -0,0 +1,101 @@
+package io.bucketeer.sdk.android.internal.model.jsonadapter
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import io.bucketeer.sdk.android.BKTException
+import io.bucketeer.sdk.android.internal.model.MetricsEventData
+import java.lang.reflect.Type
+
+class EvaluationLatencyMetricsEventAdapterFactory : JsonAdapter.Factory {
+
+  override fun create(
+    type: Type,
+    annotations: MutableSet<out Annotation>,
+    moshi: Moshi,
+  ): JsonAdapter<MetricsEventData.GetEvaluationLatencyMetricsEvent>? {
+    if (!Types.getRawType(type).isAssignableFrom(MetricsEventData.GetEvaluationLatencyMetricsEvent::class.java)) {
+      return null
+    }
+
+    return object : JsonAdapter<MetricsEventData.GetEvaluationLatencyMetricsEvent>() {
+
+      private val options: JsonReader.Options = JsonReader.Options.of("labels", "duration")
+
+      private val mapOfStringStringAdapter: JsonAdapter<Map<String, String>> =
+        moshi.adapter(
+          Types.newParameterizedType(
+            Map::class.java,
+            String::class.java,
+            String::class.java,
+          ),
+          emptySet(),
+          "labels",
+        )
+
+      private val durationAdapter: JsonAdapter<String?> = moshi.adapter(
+        String::class.java,
+        emptySet(),
+        "duration",
+      )
+
+      override fun fromJson(reader: JsonReader): MetricsEventData.GetEvaluationLatencyMetricsEvent? {
+        var labels: Map<String, String>? = null
+        var duration: Long? = null
+        reader.beginObject()
+        while (reader.hasNext()) {
+          when (reader.selectName(options)) {
+            0 -> {
+              labels = mapOfStringStringAdapter.fromJson(reader) ?: throw BKTException.IllegalStateException("unexpected type: $type")
+            }
+            1 -> {
+              duration = durationAdapter.fromJson(reader)?.protobufDurationToLong() ?: throw BKTException.IllegalStateException("unexpected type: $type")
+            }
+            -1 -> {
+              // Unknown name, skip it.
+              reader.skipName()
+              reader.skipValue()
+            }
+          }
+        }
+        reader.endObject()
+        return MetricsEventData.GetEvaluationLatencyMetricsEvent(
+          labels = labels as Map<String, String>,
+          duration = duration ?: throw BKTException.IllegalStateException("unexpected type: $type"),
+        )
+      }
+
+      override fun toJson(writer: JsonWriter, value: MetricsEventData.GetEvaluationLatencyMetricsEvent?) {
+        if (value == null) {
+          writer.nullValue()
+          return
+        }
+        writer.beginObject()
+        writer.name("labels")
+        writer.jsonValue(value.labels)
+        writer.name("duration")
+        // Kenji : because duration is in `second`, we are safe to do that
```

Comment:
Who is Kenji?

May be `NOTE` instead?

```suggestion
        // NOTE: because duration is in `second`, we are safe to do that
```

Reply:
Will change to "NOTE"